plugin/evm/vm.go

```diff
@@ -933,13 +951,68 @@ func (vm *VM) SetState(_ context.Context, state snow.State) error {
 }
 
 // initBlockBuilding starts goroutines to manage block building
-func (vm *VM) initBlockBuilding() {
+func (vm *VM) initBlockBuilding() error {
 	// NOTE: gossip network must be initialized first otherwise ETH tx gossip will not work.
 	gossipStats := NewGossipStats()
 	vm.gossiper = vm.createGossiper(gossipStats)
 	vm.builder = vm.NewBlockBuilder(vm.toEngine)
 	vm.builder.awaitSubmittedTxs()
-	vm.Network.SetGossipHandler(NewGossipHandler(vm, gossipStats))
+
+	ethTxPool, err := NewGossipEthTxPool(vm.txPool)
+	if err != nil {
+		return err
+	}
+	vm.shutdownWg.Add(1)
+	go ethTxPool.Subscribe(vm.shutdownChan, &vm.shutdownWg)
+
+	atomicMempool, err := NewGossipAtomicMempool(vm.mempool)
+	if err != nil {
+		return err
+	}
+	vm.atomicMempool = atomicMempool
+
+	vm.Network.SetGossipHandler(NewGossipHandler(
+		vm,
+		gossipStats,
+	))
```

Comment:
Can we move this back to the top to reduce the diff? I think this will make the code a little more well organized as well, so that we initialize the old gossip handler before adding the new protocol gossip handler.